src/Spice86.Core/Emulator/OperatingSystem/Dos.cs

```diff
@@ -205,6 +208,12 @@ private void OpenDefaultFileHandles(VirtualFileBase[] fileDevices) {
         }
     }
 
+    internal void GeneratePspInMemory(Configuration configuration, ushort pspSegment, string? arguments) {
+        var pspGenerator = new PspGenerator(_memory, EnvironmentVariables, MemoryManager, FileManager);
+        string dosFileSpec = $"{Path.GetFileName(configuration.Exe)}".ToUpperInvariant();
```

Comment:
I would make the exe name a parameter so that in the future when we have subprograms we can use that instead of the conf